From c682c9e2ab3466f78f16231806e8255a115b1f2c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Zolt=C3=A1n=20B=C3=B6sz=C3=B6rm=C3=A9nyi?=
 <zboszor@gmail.com>
Date: Wed, 7 Jan 2026 12:11:30 +0100
Subject: [PATCH 08/14] Use a pre-built readerwriterqueue dependency
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Use <readerwriterqueue/readerwriterqueue.h> from the external
dependency.

Do not link to libreaderwriterqueue, this dependency is
header-only.

Signed-off-by: Zoltán Böszörményi <zboszor@gmail.com>
Upstream-Status: Inappropriate [oe specific]
---
 .../cmake/external/readerwriterqueue.cmake    | 22 +------------------
 hailort/common/CMakeLists.txt                 |  1 -
 hailort/common/thread_safe_queue.hpp          |  2 +-
 3 files changed, 2 insertions(+), 23 deletions(-)

diff --git a/hailort/cmake/external/readerwriterqueue.cmake b/hailort/cmake/external/readerwriterqueue.cmake
index 403b1bc..1c5b595 100644
--- a/hailort/cmake/external/readerwriterqueue.cmake
+++ b/hailort/cmake/external/readerwriterqueue.cmake
@@ -1,23 +1,3 @@
 cmake_minimum_required(VERSION 3.11.0)
 
-include(FetchContent)
-
-FetchContent_Declare(
-    readerwriterqueue
-    GIT_REPOSITORY https://github.com/cameron314/readerwriterqueue
-    GIT_TAG 435e36540e306cac40fcfeab8cc0a22d48464509 # Version 1.0.3
-    GIT_SHALLOW TRUE
-    SOURCE_DIR ${HAILO_EXTERNAL_DIR}/readerwriterqueue-src
-    SUBBUILD_DIR ${HAILO_EXTERNAL_DIR}/readerwriterqueue-subbuild
-)
-
-# https://stackoverflow.com/questions/65527126/disable-install-for-fetchcontent
-FetchContent_GetProperties(readerwriterqueue)
-if(NOT readerwriterqueue_POPULATED)
-    FetchContent_Populate(readerwriterqueue)
-    if (NOT HAILO_EXTERNALS_EXCLUDE_TARGETS)
-        # Add readerwriterqueue as a header-only library
-        add_library(readerwriterqueue INTERFACE)
-        target_include_directories(readerwriterqueue INTERFACE ${readerwriterqueue_SOURCE_DIR})
-    endif()
-endif()
\ No newline at end of file
+find_package(readerwriterqueue REQUIRED CONFIG)
diff --git a/hailort/common/CMakeLists.txt b/hailort/common/CMakeLists.txt
index ca7d2be..3e808fe 100644
--- a/hailort/common/CMakeLists.txt
+++ b/hailort/common/CMakeLists.txt
@@ -37,7 +37,6 @@ target_include_directories(hailort_common
 target_link_libraries(hailort_common
     PUBLIC
     spdlog::spdlog
-    readerwriterqueue
     xxhash
 )
 
diff --git a/hailort/common/thread_safe_queue.hpp b/hailort/common/thread_safe_queue.hpp
index a8f05e6..cfab96b 100644
--- a/hailort/common/thread_safe_queue.hpp
+++ b/hailort/common/thread_safe_queue.hpp
@@ -25,7 +25,7 @@
 #define HAILO_UNDEF_UNIX_FLAG 
 #endif
 
-#include "readerwriterqueue.h"
+#include <readerwriterqueue/readerwriterqueue.h>
 
 #if defined(HAILO_UNDEF_UNIX_FLAG)
 #undef __unix__
-- 
2.52.0

